assets/src/components/detours/diversionPage.tsx

```diff
@@ -131,7 +142,46 @@ export const DiversionPage = ({
         </header>
 
         <div className="l-diversion-page__panel bg-light">
-          {snapshot.matches({ "Detour Drawing": "Editing" }) ? (
+          {snapshot.matches({ "Detour Drawing": "Pick Route Pattern" }) ? (
```

Comment:
Non-blocking: Would it make sense to lift the `Pick Route Pattern` and `Share Detour` machines to top level? I would expect the top-level states to basically line up with the panels on the left, and it surprised me a bit to see `Detour Drawing` as part of the condition that governs the panel that shows up _before_ you start drawing.

Reply:
Non-blocking for a few reasons, one of which is that I don't know enough about state machines / xstate to know whether this is even a good idea.

But also because I think it would be good for me or Hannah to try our hands at doing a state machine refactor like this.

And most importantly because I don't want to block merging of this (and the work that depends on this) on another style change.

Reply:
> But also because I think it would be good for me or Hannah to try our hands at doing a state machine refactor like this.

I agree with this! Maybe open a ticket and put it on the [Skate Tech Cleanup board](https://app.asana.com/0/1152340551558956/1204993290118919)?



> Non-blocking: Would it make sense to lift the `Pick Route Pattern` and `Share Detour` machines to top level? I would expect the top-level states to basically line up with the panels on the left, and it surprised me a bit to see `Detour Drawing` as part of the condition that governs the panel that shows up _before_ you start drawing.

I'm not really sure on this, I initially nested them in case we needed say, parallel states running at all times, or other modes we haven't thought of. Since there's no real easy LSP way of refactoring the `matches` calls, I wouldn't opt to do this _myself_ but I don't mind entertaining the idea and trying it if one of you would like to make that refactor.